### PR TITLE
Replace Sqs implementation with symfony/amazon-sqs-messenger

### LINKED ISSUE #29: Use symfony/amazon-sqs-messenger Transport for sending messages
I was wondering if it is an idea now symfony/amazon-sqs-messenger is released, to use the Transport from that package so this package handles only the serverless consumer side?
Does that make sense?

I doesn't seem right to keep that functionality in 2 places?

### PR DESCRIPTION
**Pull request by starred-gijs**
Fixes #29, replaces #31 (because it was easier to apply the changes again, instead of handling merge conflicts)

I have some questions, see my comments below. Also, the build failed, what do I need to fix?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,13 +25,14 @@
         "bref/bref": "^0.5.18",
         "async-aws/sns": "^1.0",
         "async-aws/sqs": "^1.2",
-        "async-aws/event-bridge": "^1.0"
+        "async-aws/event-bridge": "^1.0",
+        "symfony/amazon-sqs-messenger": "5.2.x-dev"
```
Comment (starred-gijs) [inline composer.json:29]: I will update this when 5.2 is released (or is there another way im not aware of?)

File: src/Service/Sqs/SqsTransport.php [hunk 2]
```diff
@@ -1,94 +0,0 @@
...
-            $messageId = $result->getMessageId();
-        } catch (Throwable $e) {
-            throw new TransportException($e->getMessage(), 0, $e);
-        }
-
-        if ($messageId === null) {
-            throw new TransportException('Could not add a message to the SQS queue');
-        }
-
-        return $envelope->with(new TransportMessageIdStamp($messageId));
```
Comment (starred-gijs) [inline src/Service/Sqs/SqsTransport.php:77]: The symfony/amazon-sqs-messenger does not add this `TransportMessageIdStamp`, not sure if this is needed or not?

File: tests/Resources/config/eventbridge.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ services:
         public: true
 
     my_eventbridge_consumer:
-        class: Bref\Messenger\Service\EventBridge\EventBridgeConsumer
+        class: Bref\Symfony\Messenger\Service\EventBridge\EventBridgeConsumer
```
Comment (starred-gijs) [inline tests/Resources/config/eventbridge.yaml:14]: The namespace changed, but the tests did not fail. Are these resource files in use?

File: README.md [hunk 4]
```diff
@@ -144,9 +143,29 @@ framework:
         transports:
             async: 
                 dsn: 'https://sqs.us-east-1.amazonaws.com/123456789/my-queue.fifo'
-                options: { message_group_id: com_example }
```
Comment (starred-gijs) [inline README.md:147]: This could be considered a BC break

**Review by starred-gijs — COMMENTED** [review]
The tests are fixed. Please review my other comments

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 The change log describes what is "Added", "Removed", "Changed" or "Fixed" between each release.
 
+## 0.2.0
```
Comment (starred-gijs) [inline CHANGELOG.md:5]: Should this be 0.4.0 or 1.0.0 to fix the BC break issue?

**Comment by xleliberty** [comment]
Wow i'me really thrilled to see this PR approaching the merge 🍾 
@starred-gijs thanks for this huge work !

**Comment by starred-gijs** [comment]
Hi @mnapoli @Nyholm, would you be so kind to provide your feedback on this PR? And have a look at my comments please

File: CHANGELOG.md (hunk 5 above)
Comment (Nyholm) [reply CHANGELOG.md:5]: 0.4.0

File: README.md (hunk 4 above)
Comment (Nyholm) [reply README.md:147]: It is fine to break BC if we update to 0.4.0

**Review by Nyholm — APPROVED** [review]
Cool cool. I think this looks alright. I have not tested the code though.

**Comment by starred-gijs** [comment]
Thanks for the review. 

> I have not tested the code though.

Im running this in prod since July, with decent traffic, for fifo and regular queues. Also with local consume during development no issues.

**Comment by Nyholm** [comment]
Thank you

**Comment by mnapoli** [comment]
Thanks for the contribution!

Just to be sure I didn't miss anything, except for the FIFO version there is no difference in how to setup this bridge?

**Comment by Nyholm** [comment]
Correct. No other changes. 

For the FIFO queue, you need to add a stamp instead of a hard coded config. 

@starred-gijs, please confirm.

**Comment by starred-gijs** [comment]
The differences are
1. Store all [headers in 1 MessageAttribute](https://github.com/brefphp/symfony-messenger/blob/0.3.4/src/Service/Sqs/SqsTransport.php#L46) vs [each header in own MessageAttribute](https://github.com/symfony/amazon-sqs-messenger/blob/5.x/Transport/Connection.php#L316)
To be clear: this means that the SQS records are incompatible between bref/smfony-messenger 0.3.4 and 0.4.0 if you do not use the PhpSerializer
2. Fifo config is indeed changed from hardcoded MessageGroupId to per message with AmazonSqsFifoStamp

The rest of the setup remains the same! If you have any questions or remarks, happy to answer them!

**Comment by mnapoli** [comment]
Thanks!

**Comment by Nyholm** [comment]
We should make sure to write this down in the change log before releasing 0.4.0

**Comment by starred-gijs** [comment]
im putting together a draft before end of today

**Comment by starred-gijs** [comment]
See #43

**Pull request merged**